k8s/examples/fencing/README.md

```diff
@@ -0,0 +1,114 @@
++---------------+-------+
+1 row in set (0.00 sec)
+```
+
+## Node Failure
+
+Check what node the mysql-0 pod is running on and make that node unavailable
+e.g. shutdown the node or stop the kubelet on the node. Now watch as the
+mysql-0 pod is rescheduled onto a different node.
+
+Note that if you are using the CSI driver there is a CSI helper pod that can be
+running as either a StatefulSet or as a Deployment. Given the nature of
+StatefulSets make sure that the mysql-0 pod and the storageos-statefulset-0 pod
+are not running on the same node or the volume will be unable to reattach.
+StorageOS has fixed this by using a deployment for the CSI helper rather than a
+StatefulSet. The deployment strategy is configurable using the
+[StorageOSCluster deploymentStratergy
+parameter](https://docs.storageos.com/docs/reference/cluster-operator/configuration).
+```bash
+kubectl get pods -l app=mysql -o wide
+NAME      READY   STATUS    RESTARTS   AGE    IP           NODE                           NOMINATED NODE   READINESS GATES
+client    1/1     Running   0          1m     10.244.2.4   ip-10-1-10-235.storageos.net   <none>           <none>
+mysql-0   1/1     Running   0          1m     10.244.1.6   ip-10-1-10-118.storageos.net   <none>           <none>
+
+kubectl get pods -n storageos -o wide
+NAME                        READY   STATUS    RESTARTS   AGE   IP          NODE                           NOMINATED NODE   READINESS GATES
+cli                         1/1     Running   0          50m   10.244.1.4  ip-10-1-10-118.storageos.net   <none>           <none>
+storageos-daemonset-b9f8v   3/3     Running   0          57m   10.1.10.112 ip-10-1-10-112.storageos.net   <none>           <none>
+storageos-daemonset-q64cm   3/3     Running   0          57m   10.1.10.235 ip-10-1-10-235.storageos.net   <none>           <none>
+storageos-daemonset-t75m2   3/3     Running   0          57m   10.1.10.118 ip-10-1-10-118.storageos.net   <none>           <none>
+storageos-statefulset-0     3/3     Running   0          57m   10.244.1.3  ip-10-1-10-118.storageos.net   <none>           <none>
+```
+
+Once the node is in a NotReady state you'll see that the mysql-0 pod has been
+rescheduled on a different node. In this example you can see that the MySQL
+client pod was scheduled on the same node as the mysql-0 pod and is still in
+the Terminating state. This is because a pod cannot be terminated until the
+kubelet comes back up and the pod is not rescheduled because a pod has no
+controller.
+
+```bash
+kubectl get pods -o wide
+NAME      READY   STATUS        RESTARTS   AGE   IP           NODE                           NOMINATED NODE   READINESS GATES
+client    1/1     Terminating   0          1m    10.244.2.4   ip-10-1-10-235.storageos.net   <none>           <none>
+mysql-0   1/1     Running       0          30s   10.244.1.6   ip-10-1-10-118.storageos.net   <none>           <none>
+```
+
+The pod can be force deleted and recreated so MySQL can be queried again.
```

Comment:
You can exec into mysql-0 and run `mysql -u root` if you fancy removing the recreation of the client pod.